phosh: 0.12.1 -> 0.13.1

zhaofengli opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

See upstream changelogs ([0.13.0](https://gitlab.gnome.org/World/Phosh/phosh/-/releases#v0.13.0), [0.13.1](https://gitlab.gnome.org/World/Phosh/phosh/-/releases#v0.13.1)). Notable changes include the ability to accept incoming phone calls on the lockscreen (I have not tested this; requires calls >= 41.alpha), the switch to logind to control torch brightness, among others.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

zhaofengli committed:
phosh: 0.12.1 -> 0.13.1

zhaofengli committed:
phoc: Change source URL to GNOME GitLab

tomfitzhenry commented:
Thanks @zhaofengli.

> Notable changes include the ability to accept incoming phone calls on the lockscreen (I have not tested this; requires calls >= 41.alpha)

Tested and working, despite me running calls 0.3.1.

Steps to reproduce:
1. Lock screen on pinephone
2. Call your pinephone from another phone.
3. Hear pinephone ringing.
4. Press power button on pinephone to wake up display.
5. See the calls app. Hang up.
6. Observe that the lockscreen is still up.

Yesterday (prior to using this PR) I received a call and, as usual, had to unlock to be able to answer the call.

> the switch to logind to control torch brightness

Tested and working.

tomfitzhenry commented on pkgs/applications/misc/phoc/default.nix:
```diff
@@ -39,8 +39,8 @@ in stdenv.mkDerivation rec {
-    domain = "source.puri.sm";
-    owner = "Librem5";
+    domain = "gitlab.gnome.org";
+    owner = "World/Phosh";
```
Consider using `fetchFromGitLab`'s `fetch` parameter:
```
group = "World";
owner = "Phosh";
```

archseer commented:
Can you also remove me from maintainers? I worked this a little bit but lost interest after seeing just how much GNOME it pulls in.

zhaofengli committed:
phosh: Remove archseer as maintainer

Per request: https://github.com/NixOS/nixpkgs/pull/133559#issuecomment-897343555

zhaofengli commented:
@archseer

> Can you also remove me from maintainers? I worked this a little bit but lost interest after seeing just how much GNOME it pulls in.

Sure thing! Sorry for the maintainer pings.

r-rmcgibbo commented:
Result of `nixpkgs-review pr 133559` at abf286ec run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>phosh</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `writeText`.
Near pkgs/applications/window-managers/phosh/default.nix:29:3:
```
   |
29 | , writeText
   |   ^
```
</li>
  </ul>
</details>

jtojnar commented on pkgs/applications/window-managers/phosh/default.nix:
```diff
@@ -88,7 +100,9 @@ in stdenv.mkDerivation rec {
 
   postUnpack = ''
```
`fetchFromGitLab` now supports `fetchSubmodules`: https://github.com/NixOS/nixpkgs/pull/131047

zhaofengli replied:
Cool, that makes things even simpler then :+1:

noneucat commented:
Torch is working nicely for me on the PinePhone with this PR :) Incoming calls are still behind my lockscreen, but I assume this will resolve itself once `calls` is bumped.

Mic92 commented:
How do I test this? I assume it does not work with sway?

```
$ /nix/store/izjp3kk30x9x34ixqa3gmn0593j1azcm-phosh-0.13.0/bin/phosh

(phoc:627973): phoc-wlroots-CRITICAL **: 07:50:14.125: [backend/session/logind.c:236] Failed to take control of session: Device or resource busy

(phoc:627973): phoc-wlroots-CRITICAL **: 07:50:14.127: [backend/session/direct.c:177] Could not get current tty number: Inappropriate ioctl for device

(phoc:627973): phoc-wlroots-CRITICAL **: 07:50:14.128: [backend/session/session.c:113] Failed to load session backend

(phoc:627973): phoc-wlroots-CRITICAL **: 07:50:14.128: [backend/backend.c:195] failed to start a session

(phoc:627973): phoc-wlroots-CRITICAL **: 07:50:14.128: [backend/backend.c:235] failed to start backend 'drm'

(phoc:627973): phoc-server-ERROR **: 07:50:14.128: Could not create backend
[1]    627973 trace trap (core dumped)  /nix/store/izjp3kk30x9x34ixqa3gmn0593j1azcm-phosh-0.13.0/bin/phosh
```

Mic92 commented:
Ok. Logging into a tty and running phosh seems to work.

tomfitzhenry commented:
I've been running this for >1 week now without issue.

SuperSandro2000 commented on pkgs/applications/window-managers/phosh/default.nix:
```diff
@@ -24,27 +26,20 @@
     repo = pname;
     rev = "v${version}";
-    sha256 = "048g5sp9jgfiwq6n8my4msm7wy3pdhbg0wxqxvps4m8qf8wa7ffq";
+    fetchSubmodules = true;
```
I don't think we need or want those submodules.

jtojnar replied:
Unfortunately, gvc is a copylib and packaging it separately would be too much effort. Not sure about libcall-ui, maybe once it stabilizes.

But either way, they are explicitly linked as `subproject`, not a `dependency`:

https://gitlab.gnome.org/World/Phosh/phosh/-/blob/256730a8be2346cd1748d5336e334c25e977a461/meson.build#L145

jtojnar replied:
Maybe we should start adding comments when `fetchSubmodules` is necessary.

zhaofengli replied:
Ok, added a short comment to explain the necessity.

zhaofengli commented:
Updated PR for [0.13.1](https://gitlab.gnome.org/World/Phosh/phosh/-/releases#v0.13.1) which came with the helpful addition of a "Clear all" button for notifications.
